Sources/Honeycomb/Networking/URLSessionTask.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+import OpenTelemetryApi
+
+/// Returns true if this HTTP request is from our SDK itself, so that we don't recursively capture
+/// our own requests.
+private func isOTLPRequest(_ request: URLRequest) -> Bool {
+    // Just check for the OTLP version header that's always set.
+    if let headers = request.allHTTPHeaderFields {
+        for (key, _) in headers {
+            if key == "x-otlp-version" {
```

Comment:
should we check other headers as well? honeycomb api key header or something?

Reply:
This was pretty ad-hoc, and I'm open to other suggestions, but I think it's probably best to keep it simple. I don't think we want to require the honeycomb api key, because it's not strictly necessary if the app is talking to a collector, right?

This is the only required header listed on our SDK distro spec
https://github.com/honeycombio/specs/blob/main/specs/otel-sdk-distro.md#headers

Reply:
true, I forgot about the collector scenario. Makes sense 👍